azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -157,6 +169,7 @@ func resourceArmKustoClusterCreateUpdate(d *schema.ResourceData, meta interface{
 		EnableDiskEncryption:  utils.Bool(d.Get("enable_disk_encryption").(bool)),
 		EnableStreamingIngest: utils.Bool(d.Get("enable_streaming_ingest").(bool)),
 		EnablePurge:           utils.Bool(d.Get("enable_purge").(bool)),
+		LanguageExtensions:    expandKustoClusterLanguageExtensions(d.Get("language_extensions").([]interface{})),
```

Comment:
Since the main API for this doesn't currently work, it sounds like we want to use `addLanguageExtensions` and `removeLanguageExtensions` below this, after we've set the ID to add (and remove, during an update) the language extensions if necessary?

Reply:
Required a little bit more refactoring than expected. But now it should work. Manual tests (adding, removing, re-adding) were successful. Now i'm just running the acc tests to see if also the automatic testing will work.

Of course if the REST support would work as expected it would be also a lot easier to provide this feature. :-(